umpost/um2netcdf.py

```diff
@@ -986,6 +964,54 @@ def convert_32_bit(cube):
         cube.data = cube.data.astype(np.int32)
 
 
+def fix_time_coord(cube, verbose):
+    """
+    Ensures cube has a 'time' coordinate dimension.
+
+    Coordinate dimensions are reordered to ensure 'time' is the first dimension.
+    Cubes are modified in place, although it is possible iris will return new
+    copies of cubes. UM ancillary files with no time dimension are ignored.
+
+    Parameters
+    ----------
+    cube : iris Cube object
+    verbose : True to display information on stdout
+
+    Returns
+    -------
+    A (cube, unlimited_dimensions) tuple. Unlimited dimensions is None for
+    ancillary files with no time dimension.
+    """
+    try:
+        # If time is a dimension but not a coordinate dimension, coord_dims('time')
+        # returns empty tuple
+        if tdim := cube.coord_dims('time'):
+            # For fields with a pseudo-level, time may not be the first dimension
+            if tdim != (0,):
+                tdim = tdim[0]
+                neworder = list(range(cube.ndim))
```

Comment:
```suggestion
        if (tdim,) := cube.coord_dims('time'):
            # For fields with a pseudo-level, time may not be the first dimension
            if tdim != 0:
                neworder = list(range(cube.ndim))
```
Not sure if I've got the tuple assignment correct, but the general idea is to simplify
the code to remove having to carry around a redundant part of the tuple which makes it 
messy.

Also worth making the time dimension a variable?

Reply:
The suggested tuple change breaks the binary test script around unpacking.

Then checking the aiihca.paa1jan, all `coord_dims()` are empty tuples:

```python
>>> cubes = iris.load("large_data/aiihca.paa1jan")
>>> set([c.coord_dims('time') for c in cubes])
{()}
>>> cubes[0].coord_dims('pseudo_level')  # what do other coord dims look like?
(0,)
```
I'll skip any changes around these tuple mods.